exporter/loadbalancingexporter/log_exporter.go

```diff
@@ -0,0 +1,158 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package loadbalancingexporter
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/exporter/otlpexporter"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/batchpersignal"
+)
+
+var _ component.LogsExporter = (*logExporterImp)(nil)
+
+type logExporterImp struct {
+	logger *zap.Logger
+
+	loadBalancer loadBalancer
+
+	stopped    bool
+	shutdownWg sync.WaitGroup
+}
+
+// Create new logs exporter
+func newLogsExporter(params component.ExporterCreateParams, cfg configmodels.Exporter) (*logExporterImp, error) {
+	exporterFactory := otlpexporter.NewFactory()
+
+	tmplParams := component.ExporterCreateParams{
+		Logger:               params.Logger,
+		ApplicationStartInfo: params.ApplicationStartInfo,
+	}
+
+	loadBalancer, err := newLoadBalancer(params, cfg, func(ctx context.Context, endpoint string) (component.Exporter, error) {
+		oCfg := buildExporterConfig(cfg.(*Config), endpoint)
+		return exporterFactory.CreateLogsExporter(ctx, tmplParams, &oCfg)
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &logExporterImp{
+		logger:       params.Logger,
+		loadBalancer: loadBalancer,
+	}, nil
+}
+
+func (e *logExporterImp) Start(ctx context.Context, host component.Host) error {
+	if err := e.loadBalancer.Start(ctx, host); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (e *logExporterImp) Shutdown(context.Context) error {
+	e.stopped = true
+	e.shutdownWg.Wait()
+	return nil
+}
+
+func (e *logExporterImp) ConsumeLogs(ctx context.Context, ld pdata.Logs) error {
+	var errors []error
+	batches := batchpersignal.SplitLogs(ld)
+	for _, batch := range batches {
+		if err := e.consumeLog(ctx, batch); err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	return consumererror.CombineErrors(errors)
+}
+
+func (e *logExporterImp) consumeLog(ctx context.Context, ld pdata.Logs) error {
+	traceID := traceIDFromLogs(ld)
+	if traceID == pdata.InvalidTraceID() {
+		// every log may not contain a traceID
+		// generate a random traceID so the log can be routed to a random backend
+		// but do not inject the random traceID into the log itself
+		traceID = randomTraceID()
+	}
+
+	endpoint := e.loadBalancer.Endpoint(traceID)
+	exp, err := e.loadBalancer.Exporter(endpoint)
+	if err != nil {
+		return err
+	}
+
+	le, ok := exp.(component.LogsExporter)
+	if !ok {
+		expectType := (*component.LogsExporter)(nil)
+		return fmt.Errorf("expected %T but got %T", expectType, exp)
```

Comment:
I know this is the same as the trace exporter, but could you change the message to something like: 

```suggestion
		return fmt.Errorf("unable to export logs, unexpected exporter type: expected %T but got %T", expectType, exp)
```